tests/src/app/testqgsmaptoolreshape.cpp

```diff
@@ -0,0 +1,143 @@
+
+#include "qgstest.h"
+#include "qgsapplication.h"
+#include "qgsmapcanvas.h"
+#include "qgsvectorlayer.h"
+#include "qgslinestring.h"
+#include "qgsmaptoolreshape.h"
+#include "qgisapp.h"
+
+class TestQgsMapToolReshape : public QObject
+{
+    Q_OBJECT
+  public:
+    TestQgsMapToolReshape() = default;
+
+  private slots:
+    void initTestCase(); // will be called before the first testfunction is executed.
+    void cleanupTestCase(); // will be called after the last testfunction was executed.
+    void init(); // will be called before each testfunction is executed.
+    void cleanup(); // will be called after every testfunction.
+
+    void reshapeWithBindingLine();
+
+  private:
+    QgisApp *mQgisApp = nullptr;
+};
+
+void TestQgsMapToolReshape::initTestCase()
+{
+  QgsApplication::init();
+  QgsApplication::initQgis();
+
+  // Set up the QgsSettings environment
+  QCoreApplication::setOrganizationName( QStringLiteral( "QGIS" ) );
+  QCoreApplication::setOrganizationDomain( QStringLiteral( "qgis.org" ) );
+  QCoreApplication::setApplicationName( QStringLiteral( "QGIS-TEST" ) );
+
+  QgsApplication::showSettings();
+
+  // enforce C locale because the tests expect it
+  // (decimal separators / thousand separators)
+  QLocale::setDefault( QLocale::c() );
+
+  mQgisApp = new QgisApp();
+}
+
+void TestQgsMapToolReshape::cleanupTestCase()
+{
+  QgsApplication::exitQgis();
+}
+
+void TestQgsMapToolReshape::init()
+{
+}
+
+void TestQgsMapToolReshape::cleanup()
+{
+}
+
+void TestQgsMapToolReshape::reshapeWithBindingLine()
+{
+  // prepare vector layer
+  std::unique_ptr<QgsVectorLayer> vl;
+  vl.reset( new QgsVectorLayer( QStringLiteral( "LineString?crs=epsg:4326&field=name:string(20)" ), QStringLiteral( "vl" ), QStringLiteral( "memory" ) ) );
+
+  QgsGeometry g0 = QgsGeometry::fromWkt( "LineString (0 0, 1 1, 1 2)" );
+  QgsFeature f0;
+  f0.setGeometry( g0 );
+  f0.setAttribute( 0, "polyline0" );
+
+  QgsGeometry g1 = QgsGeometry::fromWkt( "LineString (2 1, 3 2, 3 3, 2 2)" );
+  QgsFeature f1;
+  f1.setGeometry( g1 );
+  f1.setAttribute( 0, "polyline1" );
+
+  vl->dataProvider()->addFeatures( QgsFeatureList() << f0 << f1 );
+
+  // prepare canvas
+  QList<QgsMapLayer *> layers;
+  layers.append( vl.get() );
+
+  QgsCoordinateReferenceSystem srs( 4326, QgsCoordinateReferenceSystem::EpsgCrsId );
+  mQgisApp->mapCanvas()->setDestinationCrs( srs );
+  mQgisApp->mapCanvas()->setLayers( layers );
+  mQgisApp->mapCanvas()->setCurrentLayer( vl.get() );
+
+  // reshape to add line to polyline0
+  QgsLineString cl0;
+  cl0.setPoints( QgsPointSequence() << QgsPoint( 1, 2 ) << QgsPoint( 2, 1 ) );
+
+  QgsCompoundCurve curve0( *cl0.toCurveType() );
+
+  QgsMapToolReshape tool0( mQgisApp->mapCanvas() );
+  tool0.mCaptureCurve = curve0;
+
+  vl->startEditing();
+  tool0.reshape( vl.get() );
+
+  f0 = vl->getFeature( 1 );
+  QCOMPARE( f0.geometry().asJson(), QStringLiteral( "{\"type\": \"LineString\", \"coordinates\": [ [0, 0], [1, 1], [1, 2], [2, 1]]}" ) );
```

Comment:
I'd suggest using wkt to compare instead - it's a bit less fragile as there's less variability in the formatting, and is what we generally do in the test